src/crate/client/tests.py

```diff
@@ -248,7 +255,7 @@ def setUp(self):
         thread = threading.Thread(target=self.serve_forever)
         thread.daemon = True  # quit interpreter when only thread exists
         thread.start()
-        time.sleep(1)
+        time.sleep(0.5)
```

Comment:
This seems a bit arbitrary. If we wanted to make sure the server is ready I think we should use some kind of wait/try/backoff pattern like we do in `layer.py` with `wait_for_http_url`

Reply:
I can look into also adding this here.

Reply:
I've just improved this on behalf of 6472212.